iree/compiler/Codegen/LLVMCPU/test/transpose_avx2_lowering.mlir

```diff
@@ -0,0 +1,73 @@
+// RUN: iree-translate -split-input-file -iree-mlir-to-vm-bytecode-module --iree-hal-target-backends=dylib-llvm-aot -mlir-disable-threading --iree-llvm-target-cpu-features="+avx2" -print-ir-after='convert-scf-to-cf' %s 2>&1 | FileCheck %s --check-prefix=CHECK-AVX2
```

Comment:
+1 to what Hanhan says. The HAL boiler plate is annoying, but this is the best way.

Also wondering why it is looking for `avx2` directly. If you look at the `hal.executable.variant` here, it has a dictionary attribute that is a list of things that HAL propagates to the backend when it sets up the target machine. So far it has `data_layout`, `native_vector_size` and `target_triple`. You could add anything to that. I didnt know that it just gives you the CPU features as a string. Thats really annoying, with string compares everywhere. Do you need to specifically look for AVX2?

Reply:
Great, let me try the `iree-opt` approach!

There are already some examples using the attribute `cpu_features` within `hal.executable.variant` (https://github.com/google/iree/blob/3ba2dcd63c5fa285c42f141eda82526511e378d4/iree/compiler/Codegen/LLVMCPU/test/materialize_launch_configuration.mlir#L837). My implementation is just looking at that attribute to find the `+avx2` feature. We have to look for AVX2 features specifically because the transpose lowering generates AVX2 specific inline assembly (see `blendps` in the provided test).